examples/vqe.lisp

```diff
@@ -16,221 +16,300 @@
 (defparameter *number-of-qubits* 4
   "Number of qubits in the quantum virtual machine")
 
-(defparameter *reference-state* "X 0; X 1" "Reference state program (HF).")
+(defparameter *reference-state* "
+DEFCIRCUIT REFERENCE:
+    X 0
+    X 1
+"
+  "Reference state program (HF).")
 
 (defparameter *ansatz* "
-RX(pi/2) 0
-H 2
-CNOT 0 1
-CNOT 1 2
-RZ(-1.0*theta[0]) 2
-CNOT 1 2
-CNOT 0 1
-RX(-pi/2) 0
-H 2
-H 0
-RX(pi/2) 2
-CNOT 0 1
-CNOT 1 2
-RZ(1.0*theta[0]) 2
-CNOT 1 2
-CNOT 0 1
-H 0
-RX(-pi/2) 2
-RX(pi/2) 0
-RX(pi/2) 1
-RX(pi/2) 2
-H 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(-0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-RX(-pi/2) 0
-RX(-pi/2) 1
-RX(-pi/2) 2
-H 3
-H 0
-RX(pi/2) 1
-H 2
-H 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(-0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-H 0
-RX(-pi/2) 1
-H 2
-H 3
-RX(pi/2) 0
-H 1
-RX(pi/2) 2
-RX(pi/2) 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-RX(-pi/2) 0
-H 1
-RX(-pi/2) 2
-RX(-pi/2) 3
-H 0
-H 1
-H 2
-RX(pi/2) 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-H 0
-H 1
-H 2
-RX(-pi/2) 3
-H 0
-H 1
-RX(pi/2) 2
-H 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-H 0
-H 1
-RX(-pi/2) 2
-H 3
-RX(pi/2) 0
-H 1
-H 2
-H 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(-0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-RX(-pi/2) 0
-H 1
-H 2
-H 3
-H 0
-RX(pi/2) 1
-RX(pi/2) 2
-RX(pi/2) 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-H 0
-RX(-pi/2) 1
-RX(-pi/2) 2
-RX(-pi/2) 3
-RX(pi/2) 0
-RX(pi/2) 1
-H 2
-RX(pi/2) 3
-CNOT 0 1
-CNOT 1 2
-CNOT 2 3
-RZ(-0.5*theta[1]) 3
-CNOT 2 3
-CNOT 1 2
-CNOT 0 1
-RX(-pi/2) 0
-RX(-pi/2) 1
-H 2
-RX(-pi/2) 3
-RX(pi/2) 1
-H 3
-CNOT 1 2
-CNOT 2 3
-RZ(-1.0*theta[0]) 3
-CNOT 2 3
-CNOT 1 2
-RX(-pi/2) 1
-H 3
-H 1
-RX(pi/2) 3
-CNOT 1 2
-CNOT 2 3
-RZ(1.0*theta[0]) 3
-CNOT 2 3
-CNOT 1 2
-H 1
-RX(-pi/2) 3
+DEFCIRCUIT ANSATZ:
```

Comment:
Maybe we should write these in `ansatz.quil` and the state as `reference-state.quil` and use `INCLUDE` to put these in.

Reply:
I like that but I'm wary of introducing problems with paths and whatnot when distributing the program. Writing the ansatz here is not pretty but it is guaranteed to work regardless of the code's location in the file system.